component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java

```diff
@@ -346,8 +347,9 @@ private boolean isPasswordPolicyViolationError(Throwable e) {
         while (e != null) {
             if (e.getCause() instanceof PolicyViolationException) {
                 return true;
-            }
-            if (e.getCause() instanceof IdentityPasswordHistoryException) {
+            } else if (e.getCause() instanceof IdentityPasswordHistoryException) {
+                return true;
+            } else if (e.getCause() instanceof IdentityEventException) {
```

Comment:
Why we treat IdentityEventException as PasswordPolicyViolationError?

Reply:
Fixed this with specified error code in https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/30/commits/610c8fe39a202097028cc7479ce9e28ee57113aa